### ISSUE TITLE
Error in (amed + 1):(bmed - 1) : result would be too long a vector

### ISSUE THREAD

**Issue by LiyanJi-code** [chrisamiller/readDepth#6]
Hi,
when I run ReadDepth on WGBS data, the errors appeared:

Error in (amed + 1):(bmed - 1) : result would be too long a vector
Calls: new ... binParams -> calcWindParams -> fdrRate -> dividePeaks
Execution halted

How can I solve it ?
Thanks in advances.

**Comment by chrisamiller** [comment]
You're going to have to provide more information, including the exact parameters and command that you ran the tool with.

**Comment by Julian-jly** [comment]
@chrisamiller 
Sorry, I made a mistook in param file.  The problem has been solved now.
But I still want to share my parameters and commands when I run ReadDepth using WGBS data. Hope you give helpful comments, thanks.
-------------------------BEGIN----------------------------------
Step 1.
# C_to_T converted hs37d5.fa preparation,
bismark_genome_preparation --path_to_bowtie /usr/local/opt/bowtie2-2.2.8/ --verbose hs37d5.fa 

Step 2.
# To prepare the chr.map.gz and chr.dat.gz files
sh runEachChr.sh # with bin size of 1000, read length of 150bp 
# ps, I use bismark to align the chr.reads.fastq to C_to_T converted hs37d5.fa.
Step3.
#To download the entrypoint file and modify the param file.
Here is my param file:
readLength      150
fdr             0.01
overDispersion  3
gcWindowSize    1000
percCNGain      0.05
percCNLoss      0.05
chunkSize       5e6
verbose         TRUE
Step 4.
#To prepare the bed files using filtered CpG sites from BSMAP results. And the input.bam for BSMAP was generated by bismark aligner.
awk -F "\t" '{print $1"\t"$2"\t"$3}' chr.CpG.BSMAP.output > chr.bed
Step 5.
# Run the ReadDepth.r in shell
Rscript ReadDepth.r
-----------------------------END------------------------------------------

Thanks,
Best regards.

**Comment by LiyanJi-code** [comment]
@chrisamiller 
Sorry, I made a mistook in param file.  The problem has been solved now.
But I still want to share my parameters and commands when I run ReadDepth using WGBS data. Hope you give helpful comments, thanks.
-------------------------BEGIN----------------------------------
Step 1.
C_to_T converted hs37d5.fa preparation,
bismark_genome_preparation --path_to_bowtie /usr/local/opt/bowtie2-2.2.8/ --verbose hs37d5.fa 

Step 2.
To prepare the chr.map.gz and chr.dat.gz files
sh runEachChr.sh # with bin size of 1000, read length of 150bp 
ps, I use bismark to align the chr.reads.fastq to C_to_T converted hs37d5.fa.
Step3.
To download the entrypoint file and modify the param file.
Here is my param file:
readLength      150
fdr             0.01
overDispersion  3
gcWindowSize    1000
percCNGain      0.05
percCNLoss      0.05
chunkSize       5e6
verbose         TRUE
Step 4.
To prepare the bed files using filtered CpG sites from BSMAP results. And the input.bam for BSMAP was generated by bismark aligner.
awk -F "\t" '{print $1"\t"$2"\t"$3}' chr.CpG.BSMAP.output > chr.bed
Step 5.
Run the ReadDepth.r in shell
Rscript ReadDepth.r
-----------------------------END------------------------------------------
All the values in fifth column of alts.dat file were below the loss threshold. These seem weird.

Thanks,
Best regards.
